iceoryx_binding_c/include/iceoryx_binding_c/enums.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_BINDING_C_ENUMS_H
+#define IOX_BINDING_C_ENUMS_H
+
+/// @brief describes the current state of a subscriber
+enum iox_SubscribeState
+{
+    SubscribeState_NOT_SUBSCRIBED = 0,
```

Comment:
@marthtz Please take a look at the c binding readme. The reason is that we are not able to use enum classes in C. Also namespaces are not supported by C and with every enum definition the global C scope gets polluted. For instance when I have a enum like
```c
enum Bla {
   fuu,
   blub,
};

int blub = 5; // does not compile since the name blub is already reserved by Bla::blub
```

So I decided to add the enum as prefix to ease scope name pollution pain.

Reply:
That's cool! Of course I'll **R**ead **T**he **F**antastic **M**anual!